File: p0141menuadv/src/main/java/com/dovhal/p0141menuadv/MainActivity.java
package com.dovhal.p0141menuadv;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

public class MainActivity extends AppCompatActivity {
    private TextView tv;
    private TextView tv2;
    private CheckBox chb;
    private Switch sw;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        // getting view-elements
        tv = (TextView)findViewById(R.id.textView);
        tv2 = (TextView)findViewById(R.id.textView2);
        chb = (CheckBox)findViewById(R.id.chbExtMenu);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.mymenu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        menu.setGroupVisible(R.id.group1, chb.isChecked());
        return super.onPrepareOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        StringBuilder sb = new StringBuilder();
        if (item.getTitle().equals("exit")){
            this.finish();
        }
        sb.append("Item menu");
        sb.append("\r\n groupId: ").append(item.getGroupId());
        sb.append("\r\n itemId: ").append(item.getItemId());
        sb.append("\r\n order: ").append(item.getOrder());
        sb.append("\r\n title: ").append(item.getTitle());
        tv2.setText(sb.toString());

        return super.onOptionsItemSelected(item);
    }

}


// menu creation from code
//    @Override
//    public boolean onCreateOptionsMenu(Menu menu) {
//        menu.add(0, 1, 0, "add");
//        menu.add(0, 2, 0, "edit");
//        menu.add(0, 3, 3, "delete");
//        menu.add(1, 4, 1, "copy");
//        menu.add(1, 5, 2, "paste");
//        menu.add(1, 6, 4, "exit");
//        return super.onCreateOptionsMenu(menu);
//    }

File: p0141menuadv/src/main/java/com/dovhal/p0141menuadv/MainActivity.kt
package com.dovhal.p0141menuadv

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.CheckBox
import android.widget.Switch
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var tv: TextView? = null
    private var tv2: TextView? = null
    private var chb: CheckBox?  = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        // getting view-elements
        tv = findViewById<View>(R.id.textView) as TextView
        tv2 = findViewById<View>(R.id.textView2) as TextView
        chb = findViewById<View>(R.id.chbExtMenu) as CheckBox
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.mymenu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
        menu.setGroupVisible(R.id.group1, chb!!.isChecked)
        return super.onPrepareOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val sb = StringBuilder()
        if (item.title == "exit") {
            finish()
        }
        sb.append("Item menu")
        sb.append("\r\n groupId: ").append(item.groupId)
        sb.append("\r\n itemId: ").append(item.itemId)
        sb.append("\r\n order: ").append(item.order)
        sb.append("\r\n title: ").append(item.title)
        tv2!!.text = sb.toString()
        return super.onOptionsItemSelected(item)
    }
}